src/community/ogcapi/ogcapi-tiles/src/main/java/org/geoserver/ogcapi/tiles/TileJSONBuilder.java

```diff
@@ -242,12 +254,77 @@ private VectorLayerMetadata getVectorLayerMetadata(FeatureTypeInfo fti) throws I
                                 Collectors.toMap(
                                         d -> d.getLocalName(), d -> toTypeName(d.getType())));
         metadata.setFields(fields);
-
+        NumberRange<Integer> zoomRange = findLayerZoomLevel(layerInfo, subset);
+        metadata.setMinZoom(zoomRange.getMinValue());
+        metadata.setMaxZoom(zoomRange.getMaxValue());
         return metadata;
     }
 
     private String toTypeName(AttributeType type) {
         // use the same mappings used for the queryables
         return org.geoserver.ogcapi.AttributeType.fromClass(type.getBinding()).getType();
     }
+
+    // find layer's min max scale denominators and matches them with the ones in the gridset
+    // to find the corresponding min and max zoom levels
+    private NumberRange<Integer> findLayerZoomLevel(LayerInfo layerInfo, GridSubset subset)
+            throws IOException {
+        NumberRange<Double> scaleRange = CapabilityUtil.searchMinMaxScaleDenominator(layerInfo);
+
+        double minScale = scaleRange.getMinimum();
+        double maxScale = scaleRange.getMaximum();
+
+        GridSet gridSet = subset.getGridSet();
+
+        int startLevel =
+                Optional.ofNullable(subset.getMinCachedZoom()).orElse(subset.getZoomStart());
+        int endLevel = Optional.ofNullable(subset.getMaxCachedZoom()).orElse(subset.getZoomStop());
+        int limit = startLevel + gridSet.getNumLevels();
```

Comment:
Why not use endLevel here? What's the difference?

Reply:
fixed